### PR TITLE
feat: Adding subset sum in dynamic programming

### PR DESCRIPTION
**Pull request by faizan2700**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by faizan2700** [comment]
test are shown complete in github workflow of my fork. someone pls check why it is taking so much time here.

**Comment by faizan2700** [comment]
update : latex format added wherever possible in documentation comments

File: dynamic_programming/subset_sum.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
+/**
+ * @file
+ * @brief [Subset Sum DP] Implementation
+ * (https://www.geeksforgeeks.org/subset-sum-problem-dp-25)
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:4]:
```suggestion
 * @brief [Subset Sum DP](https://www.geeksforgeeks.org/subset-sum-problem-dp-25) Implementation
```

File: dynamic_programming/subset_sum.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ * the subset (that is $dp(i - 1, sum) == true?$) or obtainable when including
+ * element i to the subset (that is $sum \geq A_{i}?$ and
+ * $dp(i - 1, sum - A_{i}) == true?$) in that case we will mark $dp(i, sum)$ as
+ * true. $dp(n, V)$ will store the final answer (considering all n elements is
+ * it possible to get subset of sum $V$ ?).
+ *
+ * @author [Syed Faizan](https://github.com/faizan2700)
+ */
+
+#include <array>     /// for creating std::array
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:23]:
```suggestion
#include <array>     /// for `std::array`
```

File: dynamic_programming/subset_sum.cpp [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+#include <cassert>   /// for using assert in test function
+#include <iostream>  /// for IO operations
+#include <vector>    /// for using std::vector
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:26]:
```suggestion
#include <vector>    /// for `std::vector`
```

File: dynamic_programming/subset_sum.cpp [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+ * element i to the subset (that is $sum \geq A_{i}?$ and
+ * $dp(i - 1, sum - A_{i}) == true?$) in that case we will mark $dp(i, sum)$ as
+ * true. $dp(n, V)$ will store the final answer (considering all n elements is
+ * it possible to get subset of sum $V$ ?).
+ *
+ * @author [Syed Faizan](https://github.com/faizan2700)
+ */
+
+#include <array>     /// for creating std::array
+#include <cassert>   /// for using assert in test function
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:24]:
```suggestion
#include <cassert>   /// for `assert`
```

File: dynamic_programming/subset_sum.cpp [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+#include <vector>    /// for using std::vector
+
+/**
+ * @namespace dynamic_programming
+ * @brief Dynamic Programming algorithms
+ */
+namespace dynamic_programming {
+/**
+ * @namespace subset_sum
+ * @brief [Subset Sum DP] Implementation
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:35]:
```suggestion
 * @brief Functions for [Subset Sum DP](https://www.geeksforgeeks.org/subset-sum-problem-dp-25) Implementation
```

File: dynamic_programming/subset_sum.cpp [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+ */
+namespace subset_sum {
+/**
+ * @brief for each sum less than or equal to V calculate if it is possible
+ * to get with help of first i elements. By making use of previously
+ * calculated info subset sum can be solved in $O(n\astV)$ complexity.
+ * @tparam T size of the set
+ * @param $A$ set of the allowed values.
+ * @param $n$ array size.
+ * @param $V$ subset sum to be obtained.
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:45]: Why the `$`?

File: dynamic_programming/subset_sum.cpp [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    // 1 if obtainable, 0 otherwise.
+    return dp[n][v];
+}
+}  // namespace subset_sum
+}  // namespace dynamic_programming
+
+/**
+ * @brief Self-test implementation
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:85]:
```suggestion
 * @brief Self-test implementations
```

File: dynamic_programming/subset_sum.cpp [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+/**
+ * @brief Main function
+ * @param argc count of command line arguments (ignored)
+ * @param argv array of command line arguments (ignored)
+ * @returns 0 on exit
+ */
+int main(int argc, char *argv[]) {
+    test();  // run self-test implementation
```
Comment (Panquesito7) [inline dynamic_programming/subset_sum.cpp:113]:
```suggestion
    test();  // run self-test implementations
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
👍 Good work.

**Pull request closed without merging**